examples/flava/native/train.py

```diff
@@ -146,15 +146,19 @@ def create_model(self) -> torch.nn.Module:
 
         if self.config.training.activation_checkpointing:
             check_fn = lambda submodule: isinstance(submodule, TransformerEncoderLayer)
+            checkpoint_impl = CheckpointImpl.REENTRANT
 
-            non_reentrant_wrapper = partial(
+            if strategy == "ddp":
```

Comment:
Add a comment here explaining why we use NO_REENTRANT with DDP

Reply:
Done.